src/main/java/io/vertx/proton/impl/ProtonTransport.java

```diff
@@ -284,4 +284,9 @@ public void handle(Long event) {
       }
     }
   }
+
+  ProtonTransport setIdleTimeout(int idleTimeout) {
+    this.transport.setIdleTimeout(idleTimeout);
```

Comment:
For servers, while this will cause it to emit an idle-timeout in the Open, they wont actually enforce it is adhered to unless the client had also set an idle-timeout on its connection, due to where the checking is initiated currently (the existing support was only ensuring to send heartbeats to satisfy a peers request, not enforce its own).